Adds a Fey case study (#738)
inngest/website: pages/customers/fey.mdx

    title: 'CTO, Co-founder',
  },
  avatar: '/assets/customers/fey/dennis-brotzky.jpg',
}
export const companyURL = 'https://www.feyapp.com'
export const companyDescription = `Fey is a personal finance app that that's making making financial research more accessible and effortless for anyone.`
export const ogImage = '/assets/customers/fey/og-image-fey.png'

_Fey is a personal finance app that that's making making financial research more accessible and effortless for anyone. Built by a small team, the app uses Next.js, Inngest, and Render, leveraging serverless functions wherever appropriate._

## The Challenge: Quickly process massive datasets

Fey relies on different financial data providers for data like price history, earnings, analyst estimates, financials, statistics, among many others. At the core of this data-intensive application are data pipelines which fetch, parse, clean, sort, and store the data. Also, different pieces of data come from different providers which adds additional complexity. Most of these complicated pipelines are run on schedules and may contain many discrete steps.

![A screenshot of Fey's dashboard](/assets/customers/fey/fey-dashboard.png)

> The team a tool that can quickly orchestrate tasks and reliably trigger scheduled jobsand which would allow them to easily execute these background processes from the client through an API.

Reviewer: ```suggestion
The team a tool that can quickly orchestrate tasks and reliably trigger scheduled jobs and which would allow them to easily execute these background processes from the client through an API.
```

## Inngest: An SDK that reduces code complexity

> Initially, the team used Google Cloud Composer — a platform that, although designed for data pipelines, required everything to be written as DAGs in Python and plenty of additional customization.

Reviewer: suggestion: Remove "although" for clarity. It makes it seem that it is strange that it's DAG based b/c it's a data tool, but a lot of data tools are DAG based so it reads a little weird. Also added another small word suggestion.

```suggestion
Initially, the team used Google Cloud Composer — a platform that, designed for data pipelines, required everything to be written as DAGs in Python and considerable additional customization.
```

> As soon as companies release their data, we want to have it in Fey. **With our old tool it would take up to a day** for them to show up, which was a result of mostly complicated configuration. And **with Inngest now it's like almost instant**, maximum 30 minutes.

## Combining ingestion with OpenAI

One of the competitive advantages of Fey is how they use generative AI to enhance stock research reporting.

There are two features where Fey employs OpenAI. First is summarizing news articles to provide a “human-friendly” rundown of different types financial data.

![A screenshot of Fey's news page](/assets/customers/fey/fey-news-page.jpeg)

The second feature are SEC filings, both 10-Ks and 10-Qs. Fey generates a summary of the highlights, risks, and strengths. In this way, users don't have to deal with the cognitive load of financial jargon.

![A screenshot of Fey's news page](/assets/customers/fey/fey-news-page-2.png)

Both features use GPT-4 model via their Assistants API. Both require a complex data pipeline. For example, to achieve best results when reporting on SEC filings, each piece needs to be split into a few sections, which then are run through different prompts.

> Both features are orchestrated using Inngest's [`step.invoke()`](/docs/reference/functions/step-invoke), which allows you to asynchronously call another function and handle the result, making it perfect for complex workflows.

Reviewer: suggestion: Highlight the benefit of invoke of code re-use.
```suggestion
Both features are orchestrated using Inngest's [`step.invoke()`](/docs/reference/functions/step-invoke), which allows you to asynchronously call another function and handle the result, making it perfect for composing complex workflows with re-usable components.
```

![Inngest UI used by Fey](/assets/customers/fey/fey-inngest-ui.jpeg)

## Inngest Benefits

The adoption of Inngest resulted in several positive outcomes for Fey:

1. **Concise codebase:** Inngest's SDK is designed with developers in mind, which results in elegant and self-evident code. Moreover, given that all code lives in a single codebase now, there's no need for duplicated logic in different language codebases. 
2. **Decreased context-switching:** Connected to the previous point, it is easier to introduce new features because all code lives in the same place. Working across frontend and backend is not necessary anymore.

> 3. **50x faster execution:** When companies release earnings, Fey needs to process this data and serve it to the users, which with the previous provider could take up to a day. With Inngest, the data is available instantly (or up to 30 minutes).

Reviewer: suggestion: Felt weird to say instant then caveat with a delay, maybe we should say the new max time then mention that it's often instant?
```suggestion
3. **50x faster execution:** When companies release earnings, Fey needs to process this data and serve it to the users, which with the previous provider could take up to a day. With Inngest, the data is available within 30 minutes or most often, instantly.
```
